frontend/src/pages/Home/Home.tsx

```diff
@@ -1,24 +1,79 @@
-import { FloatButton, Layout, Typography } from 'antd'
+import { Button, FloatButton, Layout, List, Typography } from 'antd'
+import { useItems } from 'services/item_service'
 import React from 'react'
 import { Link, Outlet } from 'react-router-dom'
 
+import { useActivities, useUpdateActivity } from 'services/activities_service'
+import { usePagedResultData } from 'services/utils'
+
 export interface HomeProps {}
 
 const Home = ({}: HomeProps) => {
+
+    const { data: recentsPagedResult, isPending: recentsPending, fetchStatus: recentsStatus } = useItems(1, 5, undefined, undefined, {key: "created", order: "descend"})
+    const { data: recents} = usePagedResultData(recentsPagedResult)
+
+    const { data: upcomingPagedResult, isPending: upcomingPending, fetchStatus: upcomingStatus } = useActivities(1, 5, undefined, {completed:['false'], pending:['true']})
+    const { data: upcoming } = usePagedResultData(upcomingPagedResult)
+
+    const { data: unfinishedPagedResult, isPending: unfinishedPending, fetchStatus: unfinishedStatus } = useActivities(1, 5, undefined, {completed:['false'], pending:['false'], hasEndTime:['false']})
+    const { data: unfinished } = usePagedResultData(unfinishedPagedResult)
+
+    const activityUpdateMutation = useUpdateActivity()
+
+    const markFinished = (item:any) => {
+        activityUpdateMutation.mutate({
+            token: item.token,
+            patch: { pending: false, finished: true}
+        })
+    }
+
     return (
         <div style={{}}>
             <Typography.Title level={2}>Recently Added</Typography.Title>
-            {[1, 2, 3].map(x => (
-                <div key={x}>{x}</div>
-            ))}
-            <Typography.Title level={2}>Suggested Content</Typography.Title>
-            {[1, 2, 3].map(x => (
-                <div key={x}>{x}</div>
-            ))}
+                <List loading={recentsPending && recentsStatus !== 'idle'} size='small' dataSource={recents} renderItem={(item, index) => (
+                    <List.Item>
+                        <List.Item.Meta title={item.name}/>
+                        <Button type='primary' href={`/activities/${item.item_type}/${item.token}`}>Quick Add</Button>
```

Comment:
possibly Quick Add New Activity or something?  Mike keeps getting confused about what thing can be added, is having an add button on an item that makes an activity going to exacerbate that?

Reply:
fair. Quick Add New Activity is probably too long. will think about what it could be.

Reply:
Yea, agree

tbh the headers on all of these are going to have the same problem, we'll have to have a good think on a lot of the copy